src/compiler/builderState.ts

```diff
@@ -361,13 +426,18 @@ namespace ts {
          */
         function updateExportedModules(sourceFile: SourceFile, exportedModulesFromDeclarationEmit: ExportedModulesFromDeclarationEmit | undefined, exportedModulesMapCache: ComputingExportedModulesMap) {
             if (!exportedModulesFromDeclarationEmit) {
-                exportedModulesMapCache.set(sourceFile.resolvedPath, false);
+                exportedModulesMapCache.nonExporting.add(sourceFile.resolvedPath);

```

Comment:
Is there way to ensure that exportedModulesMapCache.exporting doesnt have the entry? Otherwise all usage needs to handle the fact that it needs to be updated everywhere. 

Also because your twoway map is specific to reference set would it make sense to use Maps with values of Set<Path> | false so that its handled inside that function and we dont have to worry about which one takes preference?

Reply:
I thought about this, but it didn't make sense to me that an entry would be both updated and deleted during the lifetime of the cache (which I think corresponds to a file change event).  If that *is* possible, then I'll update the code to keep the two collections in sync.

Reply:
But exportedModulesMapCache is also copied from previous state so it is stale thing so it can change from exporting to nonExporting

Reply:
Oh.  If it's copied from the previous state, why do consumers iterate both the cache and the previous state?  I just assumed it was a delta.

Reply:
Because currently processing one is suppose to have affected files or computing thing that are commited to actual cache after done  with that affected file.. (for API users with cancellation token to preserve the half processed data and not be in invalid state)
So currently computing one if it has entry that takes preference and ignores the entry from exportedMap...

Reply:
I couldn't find the place where the cache was initialized from the existing data, but I've rearranged things to keep the set and deleted values in sync.